common/utils.py

```diff
@@ -0,0 +1,54 @@
+from typing import Any
+from dataclasses import dataclass
+from pathlib import Path
+
+from common.consts import MINIO_MAIN_PATH, PROJECT_PATH
+from metrics.consts import MetricCollections
+
+
+def singleton(cls) -> Any:
+    """Singleton implementation in form of decorator"""
+    instances = {}
+
+    def getinstance() -> Any:
+        if cls not in instances:
+            instances[cls] = cls()
+        return instances[cls]
+
+    return getinstance
+
+
+def create_weights_path(prefix: Path, collection_name: MetricCollections, suffix: str) -> Path:
```

Comment:
maybe add docstring just for convention

Reply:
you can add method into enum, so it can be dirrectly callen on `MetricCollections` instance

Reply:
Inncluded as method in `WeightsPathGenerator` class.